src/providers/extraction_providers/extractionProviderFactory.ts

```diff
@@ -0,0 +1,15 @@
+import { IExtractionProvider } from './extractionProvider';
+import { MockExtractionProvider } from './mockExtractionProvider';
+
+interface Config {
```

Comment:
This can probably eventually be exported to form part of a larger `Config` object, so we should call this something more specific like `ExtractionConfig`